Snowflake.Data/Core/CredentialManager/Infrastructure/SFCredentialManagerFileImpl.cs

```diff
@@ -0,0 +1,157 @@
+using KeyToken = System.Collections.Generic.Dictionary<string, string>;
+
+namespace Snowflake.Data.Core.CredentialManager.Infrastructure
+{
+    internal class SFCredentialManagerFileImpl : ISnowflakeCredentialManager
+    {
+        internal const string CredentialCacheDirectoryEnvironmentName = "SF_TEMPORARY_CREDENTIAL_CACHE_DIR";
+
+        internal const string CredentialCacheDirName = ".snowflake";
+
+        internal const string CredentialCacheFileName = "temporary_credential.json";
+
+        private static readonly SFLogger s_logger = SFLoggerFactory.GetLogger<SFCredentialManagerFileImpl>();
+
+        private readonly string _jsonCacheDirectory;
+
+        private readonly string _jsonCacheFilePath;
+
+        private readonly FileOperations _fileOperations;
+
+        private readonly DirectoryOperations _directoryOperations;
+
+        private readonly UnixOperations _unixOperations;
+
+        private readonly EnvironmentOperations _environmentOperations;
+
+        public static readonly SFCredentialManagerFileImpl Instance = new SFCredentialManagerFileImpl(FileOperations.Instance, DirectoryOperations.Instance, UnixOperations.Instance, EnvironmentOperations.Instance);
+
+        internal SFCredentialManagerFileImpl(FileOperations fileOperations, DirectoryOperations directoryOperations, UnixOperations unixOperations, EnvironmentOperations environmentOperations)
+        {
+            _fileOperations = fileOperations;
+            _directoryOperations = directoryOperations;
+            _unixOperations = unixOperations;
+            _environmentOperations = environmentOperations;
+            SetCredentialCachePath(ref _jsonCacheDirectory, ref _jsonCacheFilePath);
+        }
+
+        private void SetCredentialCachePath(ref string _jsonCacheDirectory, ref string _jsonCacheFilePath)
+        {
+            var customDirectory = _environmentOperations.GetEnvironmentVariable(CredentialCacheDirectoryEnvironmentName);
+            _jsonCacheDirectory = string.IsNullOrEmpty(customDirectory) ? Path.Combine(HomeDirectoryProvider.HomeDirectory(_environmentOperations), CredentialCacheDirName) : customDirectory;
+            if (!_directoryOperations.Exists(_jsonCacheDirectory))
+            {
+                _directoryOperations.CreateDirectory(_jsonCacheDirectory);
+            }
+            _jsonCacheFilePath = Path.Combine(_jsonCacheDirectory, CredentialCacheFileName);
+            s_logger.Info($"Setting the json credential cache path to {_jsonCacheFilePath}");
+        }
+
+        internal void WriteToJsonFile(string content)
+        {
+            s_logger.Debug($"Writing credentials to json file in {_jsonCacheFilePath}");
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                _fileOperations.Write(_jsonCacheFilePath, content);
+                FileInfo info = new FileInfo(_jsonCacheFilePath);
+                FileSecurity security = info.GetAccessControl();
```

Comment:
under Windows I think we don't need the check the permissions

Reply:
Removed